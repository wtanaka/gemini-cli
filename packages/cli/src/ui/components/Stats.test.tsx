/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { render } from 'ink-testing-library';
import { describe, it, expect } from 'vitest';
import {
  StatRow,
  StatsColumn,
  DurationColumn,
  FormattedStats,
} from './Stats.js';
import { Colors } from '../colors.js';

describe('<StatRow />', () => {
  it('renders a label and value', () => {
    const { lastFrame } = render(
      <StatRow label="Test Label" value="Test Value" />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders with a specific value color', () => {
    const { lastFrame } = render(
      <StatRow
        label="Test Label"
        value="Test Value"
        valueColor={Colors.AccentGreen}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });
});

describe('<StatsColumn />', () => {
  const mockStats: FormattedStats = {
    inputTokens: 100,
    outputTokens: 200,
    toolUseTokens: 50,
    thoughtsTokens: 25,
    cachedTokens: 10,
    totalTokens: 385,
  };

  it('renders a stats column with children', () => {
    const { lastFrame } = render(
      <StatsColumn title="Test Stats" stats={mockStats}>
        <StatRow label="Child Prop" value="Child Value" />
      </StatsColumn>,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders a stats column with a specific width', () => {
    const { lastFrame } = render(
      <StatsColumn title="Test Stats" stats={mockStats} width="50%" />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders a cumulative stats column with percentages', () => {
    const { lastFrame } = render(
      <StatsColumn title="Cumulative Stats" stats={mockStats} isCumulative />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('hides the tool use row when there are no tool use tokens', () => {
    const statsWithNoToolUse: FormattedStats = {
      ...mockStats,
      toolUseTokens: 0,
    };
    const { lastFrame } = render(
      <StatsColumn title="Test Stats" stats={statsWithNoToolUse} />,
    );
    expect(lastFrame()).not.toContain('Tool Use Tokens');
  });
});

describe('<DurationColumn />', () => {
  it('renders a duration column', () => {
    const { lastFrame } = render(
      <DurationColumn apiTime="5s" wallTime="10s" />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });
});
