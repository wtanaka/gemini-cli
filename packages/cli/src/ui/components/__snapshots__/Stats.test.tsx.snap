// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<DurationColumn /> > renders a duration column 1`] = `
"Duration

API Time                                      5s
Wall Time                                    10s"
`;

exports[`<StatRow /> > renders a label and value 1`] = `"Test Label                                                                                Test Value"`;

exports[`<StatRow /> > renders with a specific value color 1`] = `"Test Label                                                                                Test Value"`;

exports[`<StatsColumn /> > renders a cumulative stats column with percentages 1`] = `
"Cumulative Stats

Input Tokens                                                                                     100
Output Tokens                                                                                    200
Tool Use Tokens                                                                                   50
Thoughts Tokens                                                                                   25
Cached Tokens                                                                              10 (2.6%)
────────────────────────────────────────────────────────────────────────────────────────────────────
Total Tokens                                                                                     385"
`;

exports[`<StatsColumn /> > renders a stats column with a specific width 1`] = `
"Test Stats

Input Tokens                                   100
Output Tokens                                  200
Tool Use Tokens                                 50
Thoughts Tokens                                 25
Cached Tokens                                   10
──────────────────────────────────────────────────
Total Tokens                                   385"
`;

exports[`<StatsColumn /> > renders a stats column with children 1`] = `
"Test Stats

Input Tokens                                                                                     100
Output Tokens                                                                                    200
Tool Use Tokens                                                                                   50
Thoughts Tokens                                                                                   25
Cached Tokens                                                                                     10
────────────────────────────────────────────────────────────────────────────────────────────────────
Total Tokens                                                                                     385
Child Prop                                                                               Child Value"
`;
